datafusion/core/src/dataframe.rs

```diff
@@ -2292,4 +2365,41 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn write_parquet_with_compression() -> Result<()> {
+        let test_df = test_table().await?;
+        let tmp_dir = TempDir::new()?;
+        let local = Arc::new(LocalFileSystem::new_with_prefix(&tmp_dir)?);
+        let local_url = Url::parse("file://local").unwrap();
+        let ctx = &test_df.session_state;
+        ctx.runtime_env().register_object_store(&local_url, local);
+
+        let output_path = "file://local/test.parquet";
+        test_df
+            .write_parquet(
+                output_path,
+                DataFrameWriteOptions::new().with_single_file_output(true),
+                Some(
+                    WriterProperties::builder()
+                        .set_compression(parquet::basic::Compression::SNAPPY)
```

Comment:
The test looks good. I wonder if it is worth looping over a list of supported compressions rather than just testing this with one codec? If the default changed to `SNAPPY` in the future then this test would not really be testing that the WriterProperties value is respected.

Reply:
I expanded the test to include all supported compression codecs. We unfortunately cannot test all compression levels for those codecs that support levels, since in general they do not include the used compression level in the file metadata. The parquet crate reader always reports the compression level as the "default" level.

Reply:
Relevant lines in parquet crate:

https://github.com/apache/arrow-rs/blob/eeba0a3792a2774dee1d10a25340b2741cf95c9e/parquet/src/file/metadata.rs#L640

https://github.com/apache/arrow-rs/blob/eeba0a3792a2774dee1d10a25340b2741cf95c9e/parquet/src/format.rs#L3495

I suppose outside of testing, there is no compelling reason for the parquet file to store the compression level that was used for each column chunk.